auth/auth_test.go

```diff
@@ -1,13 +1,64 @@
 package auth
 
 import (
+	"errors"
+	"github.com/stretchr/testify/mock"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/stakwork/sphinx-tribes/config"
 	"github.com/stretchr/testify/assert"
 )
 
+type MockParser struct {
+	mock.Mock
+}
+
+func (m *MockParser) ParseTokenString(t string) (uint32, []byte, []byte, error) {
+	args := m.Called(t)
+	return args.Get(0).(uint32), args.Get(1).([]byte), args.Get(2).([]byte), args.Error(3)
+}
+
+type MockVerifier struct {
+	mock.Mock
+}
+
+func (m *MockVerifier) VerifyAndExtract(msg, sig []byte) (string, bool, error) {
+	args := m.Called(msg, sig)
+	return args.String(0), args.Bool(1), args.Error(2)
+}
+
+type Auth struct {
+	ParseTokenString func(string) (uint32, []byte, []byte, error)
+	VerifyAndExtract func([]byte, []byte) (string, bool, error)
+}
+
+func (a *Auth) VerifyTribeUUID(uuid string, checkTimestamp bool) (string, error) {
+
+	timestamp, timeBuf, sigBuf, err := a.ParseTokenString(uuid)
+	if err != nil {
+		return "", err
+	}
+
+	if checkTimestamp {
+		now := time.Now().Unix()
+		if int64(timestamp) > now {
+			return "", errors.New("timestamp is in the future")
+		}
+		if now-int64(timestamp) > 300 {
+			return "", errors.New("too late")
+		}
+	}
+
+	pubkey, valid, err := a.VerifyAndExtract(timeBuf, sigBuf)
+	if err != nil || !valid {
+		return "", errors.New("invalid signature or verification failed")
+	}
+
+	return pubkey, nil
+}
```

Comment:
@MahtabBukhari, this function has already been declared in auth.go, why redeclare it here?

Reply:
@elraphty you are right I have updated just need to test and will push correct version.

Reply:
@elraphty @tomsmith8  Here we have to redeclare "VerifyTribeUUID"  because it is using these you method in his body 

"ParseTokenString",  "VerifyAndExtract"  for mocking these two method then we also have to use in "VerifyTribeUUID" to check the tests only for "VerifyTribeUUID" 


When we use the original "VerifyTribeUUID" then the unit tests are not going to be pass in any 



Please guide me what would be the best way now?

Reply:
You have to create a real signature, you are trying to bypass testing the original function, checkout https://github.com/stakwork/sphinx-tribes/pull/2113

Reply:
> You have to create a real signature, you are trying to bypass testing the original function, check out #2113

@elraphty Here is the correct unit test PR https://github.com/stakwork/sphinx-tribes/pull/2130